### PR TITLE
MINIFICPP-1626: Upgrade PCAP++ to a more recent release

### PR DESCRIPTION
**Pull request by martinzink**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by martinzink** [comment]
With this update we will lose one patch
https://github.com/apache/nifi-minifi-cpp/commit/0a85355f8208f7cabd81e5fee9a72075289a1b28#diff-d5fb6148c112e5a8e71e9461291971936c683b6217ecc35b2af427dfd68edf68R94-R95
volatile keyword was added to PcapLiveDevice::m_StatsThreadStarted and PcapLiveDevice::m_StopThread for some reason during SUSE and SLES12 release support.
Unfortunetly I am not sure why this was added, or that it is still needed.

File: extensions/pcap/CMakeLists.txt
Comment (adam-markovics) [inline extensions/pcap/CMakeLists.txt:28]: FetchContent_GetProperties is missing, `pcapplusplus_POPULATED` is unset, we will always go into this `if`. Could FetchContent_MakeAvailable be enough?
https://cmake.org/cmake/help/latest/module/FetchContent.html#populating-the-content

File: extensions/pcap/CMakeLists.txt
Comment (adam-markovics) [inline extensions/pcap/CMakeLists.txt:25]: Why not use GIT_REPOSITORY and GIT_TAG instead of URL and URL_HASH?

Comment (szaszm) [reply extensions/pcap/CMakeLists.txt:25]: I prefer keeping the release tarball. It's faster to download and extract than cloning a git repo.

File: extensions/pcap/CMakeLists.txt
Comment (martinzink) [reply extensions/pcap/CMakeLists.txt]: Good catch, you are right, changed it to FetchContent_MakeAvailable in https://github.com/apache/nifi-minifi-cpp/pull/1161/commits/a513ff65eff18b5f0e2394ccaf9d3d9e1a51346a

File: extensions/pcap/CMakeLists.txt
Comment (martinzink) [reply extensions/pcap/CMakeLists.txt:25]: Yeah, in this case git clone would use ~130 MB (instead of the tarball which is only ~40 MB extracted), while not providing any clear benefit.

File: extensions/pcap/CMakeLists.txt
Comment (szaszm) [inline extensions/pcap/CMakeLists.txt:32]: Could you convert the `include_directories` calls to target-oriented equivalents? If you don't mind fixing this old issue.

**Comment by szaszm** [comment]
> With this update we will lose one patch
> [0a85355#diff-d5fb6148c112e5a8e71e9461291971936c683b6217ecc35b2af427dfd68edf68R94-R95](https://github.com/apache/nifi-minifi-cpp/commit/0a85355f8208f7cabd81e5fee9a72075289a1b28#diff-d5fb6148c112e5a8e71e9461291971936c683b6217ecc35b2af427dfd68edf68R94-R95)
> volatile keyword was added to PcapLiveDevice::m_StatsThreadStarted and PcapLiveDevice::m_StopThread for some reason during SUSE and SLES12 release support.
> Unfortunetly I am not sure why this was added, or that it is still needed or not.

Looking at the change, I'm pretty sure whatever the intention was, it didn't achieve it, so I think it's safe to remove it.

File: extensions/pcap/CMakeLists.txt
Comment (adam-markovics) [reply extensions/pcap/CMakeLists.txt:25]: Okay.

**Review by adam-markovics — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Review by arpadboda — APPROVED** [review]

**Comment by szaszm** [comment]
Waiting for #1138 before this is merged

**Comment by martinzink** [comment]
#1138 added -fPIC flag to thirdparty/pcap++/3rdParty/LightPcapNg/Makefile thirdparty/pcap++/Common++/Makefile thirdparty/pcap++/Packet++/Makefile thirdparty/pcap++/Pcap++/Makefile to facilate using them in shared libs
Fortunetly this has already been in added to this new version https://github.com/seladb/PcapPlusPlus/commit/79a27431a6bebf87dc74a58d2f7d85db4b7d1fc1

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**
